### PR TITLE
Hide close items

### PR DESCRIPTION
**Pull request by bomsy**
Associated Issue: #1139

### Summary of Changes

* Added new buildMenu function and tests
* Use the buildMenu function to generate the list of menuitem that should be visible

### Testing

* [x] passes `npm test`
* [x] passes `npm run lint`

### Screenshots/Videos
hide-close

### CODE REVIEW

File: public/js/utils/tests/menu.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+
+describe("menu", () => {
+  it("should return an array of visible menu items", () => {
+    let menuItems = [
+      { item: "bla", hidden: false },
+      { item: "foo", hidden: true },
+      { item: "baa", hidden: () => true },
+      { item: "foobar" }
+    ];
+    expect(buildMenu(menuItems)[1]).to.be("foobar");
```
Comment (jasonLaster) [inline public/js/utils/tests/menu.js]: Hmm lets check yah there are three items

File: public/js/components/SourceTabs.js [hunk 2]
```diff
@@ -125,12 +125,13 @@ const SourceTabs = React.createClass({
...
-      closeTabMenuItem,
-      closeOtherTabsMenuItem,
-      closeTabsToRightMenuItem,
-      closeAllTabsMenuItem
-    ]);
+    showMenu(e, buildMenu([
+      { item: closeTabMenuItem },
+      { item: closeOtherTabsMenuItem, hidden: () => tabs.size === 1 },
+      { item: closeTabsToRightMenuItem, hidden: () =>
+         tabs.some((t, i) => t === tab && (tabs.size - 1) === i) },
```
Comment (jasonLaster) [inline public/js/components/SourceTabs.js]: Why do we need some?

Comment (bomsy) [inline public/js/components/SourceTabs.js]: Some returns a true if there's a truthy for any of the array values. i wanted a clean way of return a truthy if we hit the current tab and its the last one in the array.
Any better ideas?

Comment (jasonLaster) [inline public/js/components/SourceTabs.js]: Hmm I'm missing something. 

What do you mean by truthy in the array?

What do you mean by truthy if the current tax is the last one :)

Comment (bomsy) [inline public/js/components/SourceTabs.js]: so for every loop over the tabs items, i'm checking two things

1. `t === tab`. If the current iteration tab is the selected one.

2.  `(tabs.size - 1) === i)` . if the tab item is the last one in the array 

 If these 2 conditions are satisfied , a `true` is returned to `some` meaning that this item does not have tab items to its right.

Comment (bomsy) [inline public/js/components/SourceTabs.js]: :) hope this helps explain my thoughts behind this?

**Pull request merged**